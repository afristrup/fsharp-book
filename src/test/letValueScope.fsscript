#indent "off"

begin
printfn "1.";
let p = 3 in let p = 4 in printfn " %A" p; printfn " %A" p; // p not defined
end;

begin
printfn "2.";
let p = 3 in (let p = 4 in printfn " %A" p); printfn " %A" p;
end;

begin
printfn "3.";
let p = 3 in (let p = 4 in printfn " %A" p; printfn " %A" p);
end;

begin
printfn "4.";
let p = 3 in let p = 4 in (printfn " %A" p; printfn " %A" p);
end;

begin
printfn "5.";
let p = 3 in let p = 4 in printfn " %A" p;
printfn " %A" p;
end;

begin
printfn "6.";
let p = 3 in
let p = 4 in printfn " %A" p;
printfn " %A" p;
end;

(*begin
printfn "6a.";
let p = 3
let p = 4 in printfn " %A" p;
printfn " %A" p // p undefined
end;
*)
begin
printfn "6b.";
let p = 3 in
  let p = 4 in printfn " %A" p;
  printfn " %A" p // p undefined
end;

begin
printfn "6c.";
let p = 3 in
  let p = 4 in printfn " %A" p;
printfn " %A" p // p undefined
end;

begin
printfn "7.";
let p = 3 in                
let p = 4 in               
printfn " %A" p;
end;
//printfn " %A" p // Not OK

(*begin
printfn "7a.";
let p = 3
let p = 4 in
printfn " %A" p;
end;
begin
printfn "7b.";
let p = 3 in
let p = 4
printfn " %A" p;
end;

begin
printfn "7c.";
let p = 3;
printfn " %A" p;
let p = 4;
printfn " %A" p;
end;
*)
//printfn " %A" p // Not OK
