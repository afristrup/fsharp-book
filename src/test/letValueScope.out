> #indent "off"

  #indent "off"
  ^^^^^^^^^^^^^

/Users/sporring/repositories/fsharpNotes/src/stdin(2,1): warning FS0062: This construct is for ML compatibility. Consider using a file with extension '.ml' or '.mli' instead. You can disable this warning by using '--mlcompatibility' or '--nowarn:62'.
- 
- begin
- printfn "1.";
